### PR TITLE
Enhancement: Modelist Message Extensions

### PR DESCRIPTION
**Pull request by DanUgore**
Add message extensions for 346, 348, and 367. (Invite, Except, and Ban Lists)

### CODE REVIEW

File: src/lib/message.sjs [hunk 1]
```diff
@@ -191,6 +191,23 @@ var extensions = {
...
     },
 
+    "346": function (message) {
+        // :<server> 346 <me> <channel> <banmask> <banner> <bantime>
+        message.replyname = "RPL_INVITELIST";
+        message.channel = message.params[1];
+        message.setmask = message.params[2];
+        message.setnick = message.params[3];
+        message.settime = message.params[4];
+    },
```
Comment (Havvy) [inline src/lib/message.sjs]: Nit: Missing Newline.

File: src/lib/message.sjs [hunk 2]
```diff
@@ -191,6 +191,23 @@ var extensions = {
...
+    "348": function (message) {
+        // :<server> 348 <me> <channel> <banmask> <banner> <bantime>
```
Comment (Havvy) [inline src/lib/message.sjs]: Shouldn't the ban names be exceptmask, mode, and timestamp respectively?

File: src/lib/message.sjs [hunk 3]
```diff
@@ -191,6 +191,23 @@ var extensions = {
         message.replyname = "RPL_WHOISBOT";
     },
 
+    "346": function (message) {
+        // :<server> 346 <me> <channel> <banmask> <banner> <bantime>
+        message.replyname = "RPL_INVITELIST";
+        message.channel = message.params[1];
+        message.setmask = message.params[2];
+        message.setnick = message.params[3];
+        message.settime = message.params[4];
```
Comment (Havvy) [inline src/lib/message.sjs]: `message.setnick` should be `who`, and `message.settime` should be `timestamp` to align with `RPL_TOPICWHOTIME`.

`message.setmask` should be `.whoHostmask` I think.

Comment (DanUgore) [inline src/lib/message.sjs]: the hostmask is unrelated to the nick. So I wouldn't go with `.whoHostmask`

**Comment by Havvy** [comment]
* setnick -> setter
* settime -> timestamp
* setmask -> hostmaskPattern

**Comment by DanUgore** [comment]
Nitpicks and property names fixed

**Pull request merged**